src/accelerate/accelerator.py

```diff
@@ -1162,10 +1162,10 @@ def prepare_model(self, model: torch.nn.Module, device_placement=None):
         if self.distributed_type == DistributedType.TPU and self.state.fork_launched:
             model = xmp.MpModelWrapper(model).to(self.device)
         # torch.compile should be called last.
-        if self.state.dynamo_backend != DynamoBackend.NO:
-            import torch._dynamo as dynamo
-
-            model = dynamo.optimize(self.state.dynamo_backend.value.lower())(model)
+        if self.state.dynamo_plugin.backend != DynamoBackend.NO:
+            if not hasattr(torch, "compile"):
+                raise ValueError("Using torch.compile requires a nightly install of PyTorch.")
```

Comment:
```suggestion
                raise ValueError("Using torch.compile requires PyTorch 2.0 or higher.")
```
PyTorch 2.0 will be out very soon so let's adapt the message here.